src/sqlfluff/utils/reflow/elements.py

```diff
@@ -54,10 +69,15 @@ class ReflowBlock(ReflowElement):
     # - any:            don't enforce any spacing rules
     spacing_before: str
     spacing_after: str
+    # - None:           the default (no particular preference)
+    # - leading:        prefer newline before
+    # - trailing:       prefer newline after
+    line_position: Optional[str]
```

Comment:
Should we consider making this an enum?

Reply:
Not for now - potentially in the future. I've brought in some of the additional commits from the next PR and you'll see I'm extending these configs a little to make them more flexible.